dcmio/dcmconverter/dicom_utils: can extract value from a sub-layer of…

dgoyard opened the pull request:
… the dicom. Used for enhanced storage format.

Adding a try/except identation in case the field is missing.

Commit:
dcmio/dcmconverter/dicom_utils: can extract value from a sub-layer of the dicom. Used for enhanced storage format.
Adding a try/except identation in case the field is missing.

AGrigis commented on dcmio/dcmconverter/dicom_utils.py:
```diff
@@ -99,10 +100,22 @@ def add_meta_to_nii(nii_files, dicom_dir, prefix, dcm_tags, output_directory,
+                    content[str(name)] = str(dcmimage_temp[last_tag].value)
+                except:
+                    pass
             # > add/update content
```
118: add blank line here

AGrigis commented on dcmio/dcmconverter/dicom_utils.py:
```diff
@@ -99,10 +100,22 @@ def add_meta_to_nii(nii_files, dicom_dir, prefix, dcm_tags, output_directory,
+                try:
+                    # enhances storage, the value is burried under one or
+                    # several layer(s) of sequence
+                    dcmimage_temp = dcmimage
```
current_dataset or inner_dataset rather than dcmimage_temp

AGrigis commented on dcmio/dcmconverter/dicom_utils.py:
```diff
@@ -99,10 +100,22 @@ def add_meta_to_nii(nii_files, dicom_dir, prefix, dcm_tags, output_directory,
+                    # several layer(s) of sequence
+                    dcmimage_temp = dcmimage
+                    if len(tag) > 1:
+                        for sub_tag in tag[:-1]:
```
I prefer inner rather than sub, does it clarify what we are doing here?

AGrigis commented on dcmio/dcmconverter/dicom_utils.py:
```diff
@@ -99,10 +100,22 @@ def add_meta_to_nii(nii_files, dicom_dir, prefix, dcm_tags, output_directory,
+                    if len(tag) > 1:
+                        for sub_tag in tag[:-1]:
+                            seq_field = dcmimage_temp[sub_tag]
+                            dcmimage_temp = seq_field.value[0]
```
data_element = dcmimage_temp[inner_tag]
Need to test VR?
if data_element.VR == "SQ":
    Is it sufficient to keep the first inner dataset?
    current_dataset = data_element.value[0]
else:
    raise something

dgoyard replied:
SQ is the only element from which we can go deeper. If several tags are set for one value to extract, we assume all intermediate tags point to sequence (SQ) fields.

We take the first inner sequence. So far the information we need if enclosed in ALL inner sequences (<Sequence, length 5550, at 7FCE7A5E7C00>: explore all sequences can waste time).

AGrigis commented:
The checks have failed. From the reported error it seems that the pilot in the dcmio/test/test_dcm_to_nii.py has not been modified.

Commit:
dcmio/dcmconverter/dcm_to_nii_bbox.xml: update default value for dcm_tags
dcmio/dcmconverter/dicom_utils.py: cleanup code
dcmio/test/test_dcm_to_nii.py; update input values for dcm_tags

Commit:
dcmio/dcmconverter/dicom_utils: raise exception if an inner tag does not point to a sequence field

dgoyard commented:
adding the exception if the inner_tag does not point to a sequence (VR="SQ") field

Commit:
dcmio: functions to extract manufacturer and phase encoding direction added

Commit:
dcmio/dcmreader/dcmreader: add manufacturer scanner version extractor
